solr/contrib/ltr/src/test/org/apache/solr/ltr/feature/TestFieldValueFeature.java

```diff
@@ -93,12 +98,82 @@ public void testRanking() throws Exception {
     query.add("rows", "8");
     query.remove("rq");
     query.add("rq", "{!ltr model=popularity-model reRankDocs=8}");
-    res = restTestHarness.query("/query" + query.toQueryString());
-    System.out.println(res);
+    //res = restTestHarness.query("/query" + query.toQueryString());
+    //System.out.println(res);
     assertJQ("/query" + query.toQueryString(), "/response/docs/[0]/id=='8'");
     assertJQ("/query" + query.toQueryString(), "/response/docs/[1]/id=='7'");
     assertJQ("/query" + query.toQueryString(), "/response/docs/[2]/id=='6'");
     assertJQ("/query" + query.toQueryString(), "/response/docs/[3]/id=='5'");
   }
 
+
+  @Test
+  public void testIfADocumentDoesntHaveAFieldDefaultValueIsReturned() throws Exception {
+    SolrQuery query = new SolrQuery();
+    query.setQuery("id:42");
+    query.add("fl", "*, score");
+    query.add("rows", "4");
+
+    assertJQ("/query" + query.toQueryString(), "/response/numFound/==1");
+    assertJQ("/query" + query.toQueryString(), "/response/docs/[0]/id=='42'");
+    query = new SolrQuery();
+    query.setQuery("id:42");
+    query.add("rq", "{!ltr model=popularity-model reRankDocs=4}");
+    query.add("fl", "[fv]");
+    assertJQ("/query" + query.toQueryString(), "/response/numFound/==1");
+    assertJQ("/query" + query.toQueryString(),
+            "/response/docs/[0]/=={'[fv]':'popularity:"+FIELD_VALUE_FEATURE_DEFAULT_VAL+"'}");
+
+  }
+
+
+  @Test
+  public void testIfADocumentDoesntHaveAFieldASetDefaultValueIsReturned() throws Exception {
+
+    final String fstore = "testIfADocumentDoesntHaveAFieldASetDefaultValueIsReturned";
+
+    loadFeature("popularity42", FieldValueFeature.class.getCanonicalName(), fstore,
+            "{\"field\":\"popularity\",\"defaultValue\":42.0}");
```

Comment:
And with the defaultValue being float instead of double this would be `"42"` instead of `42.0` (that way then also 42.0 versus 42 matters not, as-is I think from memory of working on the normalizer etc. 42 would not work)